lib/policy_machine_storage_adapters/active_record/postgresql.rb

```diff
@@ -1,7 +1,77 @@
-require 'active_record/hierarchical_query' # via gem activerecord-hierarchical_query
-
 module PolicyMachineStorageAdapter
   class ActiveRecord
+    class PolicyElement
+
+      # Given a list of operation set ids and a list of operations
+      # Returns a map of operations to the given operation set ids that contain them
+      def self.filtered_operation_set_ids_by_operation(operation_set_ids, operations)
```

Comment:
@ejinotti-mdsol i don't _think_ this will return operations that either don't exist or exist but aren't in an operation set. not sure how we want to handle these but i think the wrapper function should return empty arrays for these operations.
right now 
`PM.objects_for_user_or_attribute_and_operations(operator_pe, ['read', 'frog'], direct_only: true)` 
will return
`{ 'read' => [obj, obj] }` and i think it should return `{ 'read' => [obj, obj], 'frog' => [] }` 

if you agree its prob simpler to have the wrapper function just add the operation keys if they're not there, rather than changing the sql here.

Reply:
ah, good observation.  will add a spec for this.